Sming/Components/libsodium/component.mk

```diff
@@ -0,0 +1,103 @@
+	$(LIBSODIUM_SRC)/crypto_shorthash/siphash24/shorthash_siphash24.c \
+	$(LIBSODIUM_SRC)/crypto_shorthash/siphash24/ref/shorthash_siphash24_ref.c \
+	$(LIBSODIUM_SRC)/crypto_sign/crypto_sign.c \
+	$(LIBSODIUM_SRC)/crypto_sign/ed25519/sign_ed25519.c \
+	$(LIBSODIUM_SRC)/crypto_sign/ed25519/ref10/keypair.c \
+	$(LIBSODIUM_SRC)/crypto_sign/ed25519/ref10/open.c \
+	$(LIBSODIUM_SRC)/crypto_sign/ed25519/ref10/sign.c \
+	$(LIBSODIUM_SRC)/crypto_stream/chacha20/stream_chacha20.c \
+	$(LIBSODIUM_SRC)/crypto_stream/chacha20/ref/chacha20_ref.c \
+	$(LIBSODIUM_SRC)/crypto_stream/salsa20/ref/salsa20_ref.c \
+	$(LIBSODIUM_SRC)/crypto_stream/crypto_stream.c \
+	$(LIBSODIUM_SRC)/crypto_stream/salsa20/stream_salsa20.c \
+	$(LIBSODIUM_SRC)/crypto_stream/xsalsa20/stream_xsalsa20.c \
+	$(LIBSODIUM_SRC)/crypto_verify/sodium/verify.c \
+	$(LIBSODIUM_SRC)/randombytes/randombytes.c \
+	$(LIBSODIUM_SRC)/sodium/codecs.c \
+	$(LIBSODIUM_SRC)/sodium/core.c \
+	$(LIBSODIUM_SRC)/sodium/runtime.c \
+	$(LIBSODIUM_SRC)/sodium/utils.c \
+	$(LIBSODIUM_SRC)/sodium/version.c \
+	$(LIBSODIUM_SRC)/crypto_box/curve25519xchacha20poly1305/box_curve25519xchacha20poly1305.c \
+	$(LIBSODIUM_SRC)/crypto_box/curve25519xchacha20poly1305/box_seal_curve25519xchacha20poly1305.c \
+	$(LIBSODIUM_SRC)/crypto_core/ed25519/core_ed25519.c \
+	$(LIBSODIUM_SRC)/crypto_core/ed25519/core_ristretto255.c \
+	$(LIBSODIUM_SRC)/crypto_pwhash/scryptsalsa208sha256/crypto_scrypt-common.c \
+	$(LIBSODIUM_SRC)/crypto_pwhash/scryptsalsa208sha256/scrypt_platform.c \
+	$(LIBSODIUM_SRC)/crypto_pwhash/scryptsalsa208sha256/pbkdf2-sha256.c \
+	$(LIBSODIUM_SRC)/crypto_pwhash/scryptsalsa208sha256/pwhash_scryptsalsa208sha256.c \
+	$(LIBSODIUM_SRC)/crypto_pwhash/scryptsalsa208sha256/nosse/pwhash_scryptsalsa208sha256_nosse.c \
+	$(LIBSODIUM_SRC)/crypto_scalarmult/ed25519/ref10/scalarmult_ed25519_ref10.c \
+	$(LIBSODIUM_SRC)/crypto_scalarmult/ristretto255/ref10/scalarmult_ristretto255_ref10.c \
+	$(LIBSODIUM_SRC)/crypto_secretbox/xchacha20poly1305/secretbox_xchacha20poly1305.c \
+	$(LIBSODIUM_SRC)/crypto_shorthash/siphash24/shorthash_siphashx24.c \
+	$(LIBSODIUM_SRC)/crypto_shorthash/siphash24/ref/shorthash_siphashx24_ref.c \
+	$(LIBSODIUM_SRC)/crypto_sign/ed25519/ref10/obsolete.c \
+	$(LIBSODIUM_SRC)/crypto_stream/salsa2012/ref/stream_salsa2012_ref.c \
+	$(LIBSODIUM_SRC)/crypto_stream/salsa2012/stream_salsa2012.c \
+	$(LIBSODIUM_SRC)/crypto_stream/salsa208/ref/stream_salsa208_ref.c \
+	$(LIBSODIUM_SRC)/crypto_stream/salsa208/stream_salsa208.c \
+	$(LIBSODIUM_SRC)/crypto_stream/xchacha20/stream_xchacha20.c \
+	$(LIBSODIUM_SRC)/randombytes/sysrandom/randombytes_sysrandom.c
+
+COMPONENT_INCDIRS := $(LIBSODIUM_ROOT)/src/libsodium/include
+
+# results from configure script run for --host=xtensa-lx106-elf (unused defines removed)
+COMPONENT_CFLAGS := -DSODIUM_STATIC -DHAVE_C_VARARRAYS=1 -DNATIVE_LITTLE_ENDIAN=1 -DHAVE_INLINE_ASM=1 -DHAVE_WEAK_SYMBOLS=1 -DHAVE_ALLOCA_H=1 -DHAVE_ALLOCA=1 -DCONFIGURED=1
```

Comment:
Adding  -D_GNU_SOURCE=1 to the list of COMPONENT_CFLAGS makes the compilation work also under Host (Linux in my case).